app/models/system_notification.rb

```diff
@@ -0,0 +1,3 @@
+class SystemNotification < ApplicationRecord
```

Comment:
We need active_record_state_machine or something similar to manage different states of notification

Draft -> Sent -> Read -> Archived